lib/google/gax/util.rb

```diff
@@ -42,12 +42,15 @@ module Gax
     # messages using hashes but does not allow for nested hashes to instantiate
     # nested submessages.
     #
-    # @param hash [Hash] The hash to be converted into a proto message.
+    # @param hash [Hash || Class] The hash to be converted into a proto message.
+    #   If an instance of the proto message class is given, it is returned
+    #   unchanged.
     # @param message_class [Class] The corresponding protobuf message class of
     #   the given hash.
     #
     # @return [Object] An instance of the given message class.
     def to_proto(hash, message_class)
+      return hash if hash.is_a? message_class
```

Comment:
Would be good to add a check that the hash is a hash after this line.

Reply:
Done.